L1Trigger/TrackFindingTMTT/src/Histos.cc

```diff
@@ -0,0 +1,3971 @@
+          mapBarrelLayerMinR_[layer] = r;
+          mapBarrelLayerMaxR_[layer] = r;
+        } else {
+          if (mapBarrelLayerMinR_[layer] > r)
+            mapBarrelLayerMinR_[layer] = r;
+          if (mapBarrelLayerMaxR_[layer] < r)
+            mapBarrelLayerMaxR_[layer] = r;
+        }
+      } else {
+        layer = layer % 10;
+        // Range in |z| of each endcap wheel.
+        float z = fabs(stub->z());
+        if (mapEndcapWheelMinZ_.find(layer) == mapEndcapWheelMinZ_.end()) {
+          mapEndcapWheelMinZ_[layer] = z;
+          mapEndcapWheelMaxZ_[layer] = z;
+        } else {
+          if (mapEndcapWheelMinZ_[layer] > z)
+            mapEndcapWheelMinZ_[layer] = z;
+          if (mapEndcapWheelMaxZ_[layer] < z)
+            mapEndcapWheelMaxZ_[layer] = z;
+        }
+      }
+    }
+
+    // Determine Range in (r,|z|) of each module type.
+
+    for (const Stub* stub : vStubs) {
+      float r = stub->r();
+      float z = fabs(stub->z());
+      unsigned int modType = stub->digitalStub().moduleType();
+      // Do something ugly, as modules in 1-2nd & 3-4th endcap wheels are different to those in wheel 5 ...
+      // And boundary between flat & tilted modules in barrel layers 1-3 varies in z.
+      if (stub->barrel() && stub->layerId() == 1) {  // barrel layer 1
+        if (mapExtraAModuleTypeMinR_.find(modType) == mapExtraAModuleTypeMinR_.end()) {
+          mapExtraAModuleTypeMinR_[modType] = r;
+          mapExtraAModuleTypeMaxR_[modType] = r;
+          mapExtraAModuleTypeMinZ_[modType] = z;
+          mapExtraAModuleTypeMaxZ_[modType] = z;
+        } else {
+          if (mapExtraAModuleTypeMinR_[modType] > r)
+            mapExtraAModuleTypeMinR_[modType] = r;
+          if (mapExtraAModuleTypeMaxR_[modType] < r)
+            mapExtraAModuleTypeMaxR_[modType] = r;
+          if (mapExtraAModuleTypeMinZ_[modType] > z)
+            mapExtraAModuleTypeMinZ_[modType] = z;
+          if (mapExtraAModuleTypeMaxZ_[modType] < z)
+            mapExtraAModuleTypeMaxZ_[modType] = z;
+        }
+      } else if (stub->barrel() && stub->layerId() == 2) {  // barrel layer 2
+        if (mapExtraBModuleTypeMinR_.find(modType) == mapExtraBModuleTypeMinR_.end()) {
+          mapExtraBModuleTypeMinR_[modType] = r;
+          mapExtraBModuleTypeMaxR_[modType] = r;
+          mapExtraBModuleTypeMinZ_[modType] = z;
+          mapExtraBModuleTypeMaxZ_[modType] = z;
+        } else {
+          if (mapExtraBModuleTypeMinR_[modType] > r)
+            mapExtraBModuleTypeMinR_[modType] = r;
+          if (mapExtraBModuleTypeMaxR_[modType] < r)
+            mapExtraBModuleTypeMaxR_[modType] = r;
+          if (mapExtraBModuleTypeMinZ_[modType] > z)
+            mapExtraBModuleTypeMinZ_[modType] = z;
+          if (mapExtraBModuleTypeMaxZ_[modType] < z)
+            mapExtraBModuleTypeMaxZ_[modType] = z;
+        }
+      } else if (!stub->barrel() && (stub->layerId() % 10 == 1 || stub->layerId() % 10 == 2)) {  // endcap wheel 1-2
```

Comment:
Only a dummy version of class Histos, with no implementation, will be included in the updated PR. (This will allow us to replace it will the full implementation for single-threaded L1 track development work, but will avoid the problem of histograms preventing multithreaded EDProducers for CMSSW official code)